numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -221,33 +221,33 @@ npy_PyFile_Dup2(PyObject *file, char *mode, npy_off_t *orig_pos)
                         "Getting a FILE* from a Python file object failed");
     }
 
+    ret = PyObject_CallMethod(file, "seekable", NULL);
+    if (ret == NULL){
+      PyErr_SetString(PyExc_IOError,
+                      "could not determine if the file is seekable");
+      fclose(handle);
+      return NULL;
+    }
+    seekable = PyObject_IsTrue(ret);
+    Py_DECREF(ret);
+    if (seekable == -1){
+      PyErr_SetString(PyExc_IOError,
+                      "could not determine if the file is seekable");
```

Comment:
Again, probably preferable to let the exception propagate rather than replace it with your own